astroid/builder.py

```diff
@@ -264,7 +264,7 @@ def delayed_assattr(self, node):
 
 
 def build_namespace_package_module(name, path: str) -> nodes.Module:
-    return nodes.Module(name, doc="", path=path, package=True)
+    return nodes.Module(name, doc="", path=[path], package=True)
```

Comment:
This is probably a true bug fix. If `path` was set as `str`, `path[0]` would access just the first char and not the complete string. Can you create a separate PR for it, ideally with changelog entry. We should also test how this effects existing code.

Reply:
If this is a bug fix maybe we can put it in 2.8.6.

Reply:
I looked into it. I don't think `path: str` is correct here. Looking at how this is called the `path` always comes from `ModuleSpec.submodule_search_locations` which is always a `List[str]`. 

You can run the test `test_nested_namespace_import` to debug this.

Thus, this change is actually incorrect and the typing should be updated. I will push a new commit.

Reply:
That does make sense! I guess otherwise namespace packages would be broken completely.